File: JavaScript/bubblesort/bubblesort.js
// Variaveis
const NUM = 10000

const fs = require("fs")

function bubbleSort(items) {
    const inicio = Date.now();
    for (var i = 0; i < NUM; i++) { 
        for (var j = 0; j < (NUM - i - 1); j++) { 
            if(items[j] > items[j+1]) {
                var tmp = items[j]; 
                items[j] = items[j+1]; 
                items[j+1] = tmp; 
            }
        }
    }
    const fim = Date.now();
    let tempoTotal = (fim - inicio)/1000;
    return (tempoTotal)
}
function insertArray(array){
    for(let i = 0; i < NUM; i++){
        array[i] = Math.floor(Math.random() * 10000);
    }
}

function removeArray(array){
    for(let i = 0; i < NUM; i++){
        array.pop();
    }
}
var stream = fs.createWriteStream("resultado-bubblesort.csv");
stream.once('open', function(fd) {});

let tempo = 0
for(let k = 0; k < 1000; k++){
    var array = new Array(NUM);
    insertArray(array);
    tempoIndividual = bubbleSort(array)
    tempo += tempoIndividual;
    removeArray(array);
    
    stream.write((tempoIndividual*1000)+"\n");
    //console.log(`${tempo}`);
}
stream.end();
console.log(`O tempo total decorrido foi de ${tempo} segundos`);


 

File: Python/bubblesort/bubblesort.py
import random
import time

def bubbleSort(vet) :
    n=len(vet)
    for i in range(n-1):
        for j in range(n-i-1):
            if(vet[j] > vet[j + 1]):
                aux = vet[j]
                vet[j] = vet[j+1]
                vet[j+1] = aux
    
tempoMedia=0.0
arq = open("tempo-bubblesort.csv","w")

n=10000

for i in range(100):

    vet= random.sample(range(n),n)
    inicio = time.time()
    bubbleSort(vet)
    fim = time.time()
    total = (fim-inicio)*1000#milissegundos
    
    arq.write(str(total)+"\n")
    vet.clear()
    tempoMedia = tempoMedia+total


tempoMedia = tempoMedia/100 
print(tempoMedia)
arq.close()







    